xclim/indices/_multivariate.py

```diff
@@ -447,21 +499,59 @@ def heat_wave_max_length(
     characterize the occurrence of hot weather events that can result in adverse health outcomes for Canadian
     communities (Casati et al., 2013).
 
-    In Robinson (2001), the parameters would be `thresh_tasmin=27.22, thresh_tasmax=39.44, window=2` (81F, 103F).
+    The "consecutive" mode comes from Robinson (2001), where the parameters would be `thresh_tasmin=27.22, thresh_tasmax=39.44, window=2` (81F, 103F).
+
+    The "mean" mode comes from Lebel, Bustinza & Dubé (2017).
 
     References
     ----------
     Casati, B., A. Yagouti, and D. Chaumont, 2013: Regional Climate Projections of Extreme Heat Events in Nine Pilot
     Canadian Communities for Public Health Planning. J. Appl. Meteor. Climatol., 52, 2669–2698,
     https://doi.org/10.1175/JAMC-D-12-0341.1
 
+    Lebel, G., Bustinza, R. & Dubé, M. (2017). Analyse des impacts des vagues régionales de chaleur extrême sur la santé
+    au Québec de 2010 à 2015 : changements climatiques. Québec: INSPQ, Institut national de santé publique du Québec.
+    https://www.inspq.qc.ca/publications/2221
+    
     Robinson, P.J., 2001: On the Definition of a Heat Wave. J. Appl. Meteor., 40, 762–775,
     https://doi.org/10.1175/1520-0450(2001)040<0762:OTDOAH>2.0.CO;2
     """
+    if mode == "consecutive" and not np.isscalar(window):
+        raise ValueError(
+            "Window as sequence of weights is only valid for the 'mean' mode."
+        )
+
     thresh_tasmax = utils.convert_units_to(thresh_tasmax, tasmax)
     thresh_tasmin = utils.convert_units_to(thresh_tasmin, tasmin)
 
-    cond = (tasmin > thresh_tasmin) & (tasmax > thresh_tasmax)
+    if mode == "consecutive":
+        cond = (tasmin > thresh_tasmin) & (tasmax > thresh_tasmax)
+    elif mode == "mean":
+        if not np.isscalar(window):
+            weights = np.array(window)
+            window = len(window)
+            if weights.sum() != 1:
+                raise ValueError("Weights used for the window must sum to 1.")
+        else:
+            weights = np.array([1 / window] * window)
+        cond = (
+            _rolling(
+                tasmin,
+                window=window,
+                dim="time",
+                mode=lambda x, axis: np.sum(x * weights, axis=axis),
+            )
+            > thresh_tasmin
+        ) & (
+            _rolling(
+                tasmax,
+                window=window,
+                dim="time",
+                mode=lambda x, axis: np.sum(x * weights, axis=axis),
+            )
+            > thresh_tasmax
+        )
+
     group = cond.resample(time=freq)
     max_l = group.apply(rl.longest_run, dim="time")
     return max_l.where(max_l >= window, 0)
```

Comment:
Line 557. Same comment as before ... I think the run_length window condition would already met when mode=='mean'

Reply:
You are right! That adds the strange case where we have `mode='mean'` and `window=N`, but a max length under N. But hey, not our fault, @inspq.

Reply:
> You are right! That adds the strange case where we have `mode='mean'` and `window=N`, but a max length under N. But hey, not our fault, @Inspq.

We could potentially also add N-1 to max_l when mode='mean'?   

Does the INSPQ report talk about  heat wave length definitions in their report?

Reply:
Thinking out loud here not sure which option makes more sense... 

On the first day when the the N-day moving window meets heatwave threshold conditions is heat wave length equal to N or equal to 1?

@aulemahal @huard opinions?    I'm thinking more and more that it should be N....

Reply:
I thought I understood that if the N-day moving window qualified as a heatwave, it still was a 1-day heatwave. But, rereading the INSPQ report, the shortest reported heatwaves all have 3 days... Does that mean 1 = 3? The, adding N-1 to all found lengths would make sense.
However, neither the INSPQ or both its sources for this indicator discuss the "length" of the heat waves.

Reply:
I think the output from both modes should be comparable. So if for consecutive if we get 3, we should get 3 for the mean mode. 

Still unsure whether this should be an independent indicator. There are probably other heatwave definitions out there. I think it will be easier to maintain and test a set of simple independent indicators than one big indicator with lots of options.

Reply:
The more I look into the reports, the less sure I am about what they mean. Here is an excerpt from the 2015 report:

> Ainsi, une vague régionale de chaleur extrême est définie ainsi : au minimum 3 jours consécutifs pendant lesquels les moyennes mobiles des températures maximales et minimales,  observées aux stations météorologiques de référence des RSS, atteignent les seuils de chaleur extrême préalablement définis.

Should we understand here that the threshold is taken on the N-window running means, but that we **still need** N consecutive days where the condition is respected? That would mean, that we only need to add a rolling call, but all the rest stays the same.

Reply:
> I think the output from both modes should be comparable. So if for consecutive if we get 3, we should get 3 for the mean mode.
> 
> Still unsure whether this should be an independent indicator. There are probably other heatwave definitions out there. I think it will be easier to maintain and test a set of simple independent indicators than one big indicator with lots of options.

Our original thinking was that there are already 3 'types' of heatwave calculations (frequency, max_length, and now total days) repeating each type with a new definition we end up with 6 indicators which starts to be confusing